tests/integration_tests/modules/test_cli.py

```diff
@@ -36,19 +36,26 @@
 
 
 @pytest.mark.user_data(VALID_USER_DATA)
-def test_valid_userdata(client: IntegrationInstance):
-    """Test `cloud-init schema` with valid userdata.
+class TestValidUserData:
+    def test_schema_status(self, client: IntegrationInstance):
+        """Test `cloud-init schema` with valid userdata.
 
-    PR #575
-    """
-    result = client.execute("cloud-init schema --system")
-    assert result.ok
-    assert "Valid schema user-data" in result.stdout.strip()
-    result = client.execute("cloud-init status --long")
-    assert 0 == result.return_code, (
-        f"Unexpected exit {result.return_code} from cloud-init status:"
-        f" {result}"
-    )
+        PR #575
+        """
+        result = client.execute("cloud-init schema --system")
+        assert result.ok
+        assert "Valid schema user-data" in result.stdout.strip()
+        result = client.execute("cloud-init status --long")
+        assert 0 == result.return_code, (
+            f"Unexpected exit {result.return_code} from cloud-init status:"
+            f" {result}"
+        )
+
+    @pytest.mark.parametrize("mode", ["init", "config", "final"])
```

Comment:
These tests need to use `class_client` fixture instead of `client` fixture to avoid launching 4 separate instances for every parametrized integration test.